ticketutil/bugzilla.py

```diff
@@ -63,18 +115,24 @@ def _create_requests_session(self):
         We're using a Session to persist cookies across all requests made from the Session instance.
         :return s: Requests Session.
         """
-        if self.auth == 'kerberos':
+        #if self.auth == 'kerberos':
+        if isinstance(self.auth, KerberosAuth):
             return super(BugzillaTicket, self)._create_requests_session()
 
         # If basic authentication is passed in, generate a token to be used in all requests.
         try:
             s = requests.Session()
-            r = s.get(self.auth_url, params=self.credentials, verify=False)
-            r.raise_for_status()
-            resp = r.json()
-            self.token = resp['token']
-            logging.debug("Create requests session: Status Code: {0}".format(r.status_code))
-            logging.info("Successfully authenticated to {0}.".format(self.ticketing_tool))
+            
+            if isinstance(self.auth, UPAuth):
+                r = s.get(self.auth_url, params=self.credentials, verify=False)
+                r.raise_for_status()
+                resp = r.json()
+                self.auth = TokenAuth(resp["token"])
```

Comment:
I'm a little confused with the difference between UPAuth and TokenAuth. Would a user ever use TokenAuth? Or would a user just use UPAuth, which generates a token? If the latter is true, then I would get rid of the separate TokenAuth class, and just have a function in UPAuth that generates this token.